common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1082,6 +1096,43 @@ static CollationMeta loadCollationMeta(CollationIdentifier collationIdentifier)
       }
     }
 
+    /**
+     * Collation that is a result of two different non-explicit collation.
+     */
+    private static class IndeterminateCollation extends Collation {
+
+      IndeterminateCollation() {
+        super(
+          "null",
```

Comment:
Observation: Using named parameters here and below would improve readability, though I acknowledge that it may be a thing of personal preference.

Reply:
Java doesn't support named parameters unfortunately, however intellij is smart enough to add them in the UI

<img width="303" alt="image" src="https://github.com/user-attachments/assets/bed0f55e-706a-4e9f-b5f2-8cf7d2b084b2" />